rest-service/manager_rest/utils.py

```diff
@@ -25,15 +25,21 @@
 from manager_rest import constants, config, manager_exceptions
 
 
+def check_unauthenticated_endpoint():
+    """Is the accessed request unauthenticated?"""
+    if request.endpoint is None:
+        return False
+    request_endpoint = _endpoint_strip_version(request.endpoint)
+    return request_endpoint in constants.UNAUTHENTICATED_ENDPOINTS
```

Comment:
Can't this be determined by seeing which endpoint flask thinks it's routing to, given that said endpoint is not descended from securedresource? (just thinking to avoid having to remember to add things to a list in the unlikely event that we add more unauthed endpoints)

Reply:
Right, I thought about that, but I didn't find a good way to do this.

First: at this point, we don't yet know the endpoint directly. We could figure it out, like this:
```python
# this returns the .as_view method of the resource
viewfunc = current_app.view_functions[request.endpoint]
# now this gives us the resource
view_class = viewfunc.view_class

if isinstance(view_class, SecuredResource): ...
```

but I don't quite like this, because:
1. I'm not sure checking for SecuredResource to be the right place. Ideally we'd check for the existence of the authenticate decorator, but there's no good way to do that, especially in face of...
2. this then assumes all our viewfunctions are indeed descended from Resource, and are class-based indeed. That's not inevitable, we might have eventually some simple function-based views
3. the obvious way of writing this `if` is unauthenticated-default. `return isinstance(x, SecuredResource)` means that the default, eg. if the view is just a function, we default to unauthenticated. I think auth here should be opt-out, not opt-in.

And given how critical this is (failure here is catastrophic, you allow unauthenticated access!!), I'm not all that averse to a very-explicit "just a list of endpoints", even at the cost of increased burden while implementing new unauthenticated endpoints

Reply:
We could dynamically build the list of endpoints when something has the authenticate decorator added I think?

But yeah, fair point, given the impact of failure I agree, let's make it very explicit.